content/en/tech/ssh-agent-does-not-automatically-load-passphrases-on-the-osx-sierra-keychain.md

```diff
@@ -0,0 +1,81 @@
+---
+type:               "post"
+title:              "SSH-agent does not automatically load passphrases on the OSX Sierra keychain during startup"
+date:               "2016-10-14"
+publishdate:        "2016-10-14"
+draft:              false
+slug:               "ssh-agent-does-not-automatically-load-passphrases-on-the-osx-sierra-keychain"
+description:        "SSH-agent does not automatically load passphrases on the OSX Sierra keychain during startup."
+
+language:           "en"
+thumbnail:          "/images/posts/thumbnails/badass_vader.jpg"
+header_img:         "/images/posts/headers/php_elao_code.jpg"
+tags:               ["SSH", "OSX"]
+categories:         ["Dev", "Tech", "OSX"]
+
+author_username:    "gfaivre"
+---
+
+Some of us encountered an issue after upgrading to Mac OS Sierra.
+Indeed right after it our ssh keys (**with passphrases**) were not forwarded to the remote hosts anymore.
+
+In previous versions of mac OSX, `ssh-agent` used to remember the passphrases for the keys added to the keychain (with `ssh-add -K`) and after a reboot (or logout/login), it automatically picked up the passphrases from the keychain with no extra step and it was perfect !
+
+Unfortunately after upgrading to Sierra this way no longer works and command `ssh-add -K` in Sierra no longer saves SSH keys in OS's keychain.
+
+__Apple Developer stated:__
+
+> Engineering has determined that this issue behaves as intended based on the following information:
+That’s expected. We re-aligned our behavior with the mainstream OpenSSH in this area.
+You can fix this pretty easily by running ssh-add -A in your rc script if you want your keys to always be loaded.
+
+We still can login to remote hosts via ssh **BUT** ssh keys with passphrase are not forward to the host which makes the agent somewhat useless.
+
+# Doing it manually
+
+If you are fine with the behavior and don't want to store your passphrases into keychain you can do it in the old way by manually adding keys to the agent:
+
+```
+ssh-add -K /path/to/private/key
+```
+and to add identities to the agent using any passphrases stored in your keychain.
+
+```
+ssh-add -A
+```
+
+# Using ssh config file
+
+A permanent (and probably the most "cleaner" way) workaround to this behavior consist in using the new SSH option `AddKeysToAgent` option as following in your `.ssh/config` file
```

Comment:
Not sure, there's 2 things IMOO "a permanent workaround" AND "the cleanest way" @mb3rnard @xavier-rdo ?